### PR TITLE
IGAPP-910: Linting

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Review by SteffiStuffel — COMMENTED** [review]
Now even more linting warnings... For me having this amount of warnings makes them completely useless, as if my changes will add a few warning, i wouldn't even recognize as they get lost in the shuffle.
Any ideas how to deal with this?

**Comment by klinzo** [comment]
> Now even more linting warnings... For me having this amount of warnings makes them completely useless, as if my changes will add a few warning, i wouldn't even recognize as they get lost in the shuffle. Any ideas how to deal with this?

I only run `yarn lint --quiet` all the time. Which warnings are new? I thought I did not add any additional warning rules :thinking:

File: native/package.json [hunk 1]
```diff
@@ -97,9 +97,12 @@
     "react-navigation-header-buttons": "^9.0.1",
     "react-redux": "^7.2.6",
     "redux": "^4.1.2",
+    "redux-devtools-extension": "^2.13.8",
```
[the remark this answers is not in the document]
Comment (klinzo) [reply native/package.json:100]: Kind of as we use them in production code but only enable them if the build is not a developer friendly build. At least the packages say they should be installed normally:
* https://www.npmjs.com/package/redux-devtools-extension
* https://github.com/jk-gan/redux-flipper#get-started
* https://github.com/agiledigital/typed-redux-saga#installation
* https://github.com/google/schema-dts#usage

But if anyone has more insights in whether this is correct or not let me know.

**Comment by SteffiStuffel** [comment]
> > Now even more linting warnings... For me having this amount of warnings makes them completely useless, as if my changes will add a few warning, i wouldn't even recognize as they get lost in the shuffle. Any ideas how to deal with this?
> 
> I only run `yarn lint --quiet` all the time. Which warnings are new? I thought I did not add any additional warning rules thinking

So also ignoring warnings^^
I don't know, but previously it was ~70 warnings now it is ~100 warnings

**Comment by klinzo** [comment]
> So also ignoring warnings^^ I don't know, but previously it was ~70 warnings now it is ~100 warnings

Created https://issues.tuerantuer.org/browse/IGAPP-912. Hmm only kind of. When I run linting for all files, I ignore them. When I see warnings in the code/files I am working on I try to fix them (which is not always possible, e.g. for non-null assertions). So I think it is not that bad and the issue should be discussed.

File: api-client/src/endpoints/__tests__/createTrackingEndpoint.spec.ts [hunk 2]
```diff
@@ -24,25 +24,22 @@ describe('createTrackingEndpoint', () => {
     timestamp: '2020-01-20T00:00:00.000Z'
   }
   it('should throw fetch error if fetch fails', async () => {
-    ;(fetch as unknown as FetchMock).mockRejectOnce(() =>
-      Promise.reject(new Error('Das Internet ist kaputt!!!1!!!11elf!'))
+    const error = new Error('Das Internet ist kaputt!!!1!!!11elf!')
+    ;(fetch as unknown as FetchMock).mockRejectOnce(() => Promise.reject(error))
```
Comment (Taggotty) [inline api-client/src/endpoints/__tests__/createTrackingEndpoint.spec.ts:28]: just change the import to `import fetch from 'jest-fetch-mock'` them you don't need to parse this object

**Review by Taggotty — APPROVED** [review]
Maybe you could disable the warning Test has no assertions. I think we didn't have this one before and it occures quite frequently, when we used helper functions inside tests.

**Review by Taggotty — APPROVED** [review]

**Review by SteffiStuffel — APPROVED** [review]

**Pull request merged**